src/mono/dbi/cordb-process.cpp

```diff
@@ -0,0 +1,657 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+// File: CORDB-PROCESS.CPP
+//
+
+#include <cordb-appdomain.h>
+#include <cordb-assembly.h>
+#include <cordb-breakpoint.h>
+#include <cordb-code.h>
+#include <cordb-eval.h>
+#include <cordb-function.h>
+#include <cordb-process.h>
+#include <cordb-thread.h>
+#include <cordb.h>
+
+using namespace std;
+
+CordbProcess::CordbProcess(Cordb* cordb) : CordbBaseMono(NULL)
+{
+    m_pAppDomainEnum = NULL;
+    m_pBreakpoints   = new ArrayList();
+    m_pThreads       = new ArrayList();
+    m_pFunctions     = new ArrayList();
+    m_pModules       = new ArrayList();
+    appDomains       = new ArrayList();
+    m_pPendingEval   = new ArrayList();
+    this->m_pCordb   = cordb;
+    m_bIsJustMyCode  = false;
+}
+
+CordbProcess::~CordbProcess()
+{
+    if (m_pAppDomainEnum)
+        m_pAppDomainEnum->InternalRelease();
+    DWORD i = 0;
+    while (i < m_pBreakpoints->GetCount())
+    {
+        CordbFunctionBreakpoint* breakpoint = (CordbFunctionBreakpoint*)m_pBreakpoints->Get(i);
+        if (breakpoint)
+            breakpoint->InternalRelease();
+        i++;
+    }
+    i = 0;
+    while (i < m_pThreads->GetCount())
+    {
+        CordbThread* thread = (CordbThread*)m_pThreads->Get(i);
+        thread->InternalRelease();
+        i++;
+    }
+    i = 0;
+    while (i < m_pFunctions->GetCount())
+    {
+        CordbFunction* function = (CordbFunction*)m_pFunctions->Get(i);
+        function->InternalRelease();
+        i++;
+    }
+    i = 0;
+    while (i < appDomains->GetCount())
+    {
+        CordbAppDomain* appdomain = (CordbAppDomain*)appDomains->Get(i);
+        appdomain->InternalRelease();
+        i++;
+    }
+    i = 0;
+    while (i < m_pModules->GetCount())
+    {
+        CordbModule* module = (CordbModule*)m_pModules->Get(i);
+        module->InternalRelease();
+        i++;
+    }
+    i = 0;
+    while (i < m_pPendingEval->GetCount())
+    {
+        CordbEval* eval = (CordbEval*)m_pPendingEval->Get(i);
+        if (eval)
+            eval->InternalRelease();
+        i++;
+    }
+    delete m_pBreakpoints;
+    delete m_pThreads;
+    delete m_pFunctions;
+    delete m_pModules;
+    delete appDomains;
+    delete m_pPendingEval;
+    delete conn;
+}
+
+void CordbProcess::CheckPendingEval()
+{
+    DWORD i = 0;
+    while (m_pPendingEval && i < m_pPendingEval->GetCount())
+    {
```

Comment:
I would use return early and `continue` to avoid too many nested levels

```cpp

    if (!m_pPendingEvel)
          return;
    
    for (DWORD i = 0; i < m_pPendingEval->GetCount(); i++)
    {
        CordbEval* eval = (CordbEval*)m_pPendingEval->Get(i);
        if (!eval)
              continue;
        ReceivedReplyPacket* recvbuf = conn->GetReplyWithError(eval->GetCommandId());
        if (!recvbuf)
              continue;

          eval->EvalComplete(recvbuf->Buffer());
          eval->InternalRelease();
          dbg_lock();
          m_pPendingEval->Set(i, NULL);
          dbg_unlock();
   }
```

Reply:
Done.